src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java

```diff
@@ -418,6 +422,58 @@ public void deleteTemplate(String projectId, String templateName) throws IOExcep
         return instanceTemplates;
     }
 
+    /**
+     * Creates persistent disk snapshot for Compute Engine instance.
+     * This method blocks until the operation completes.
+     *
+     * @param projectId Google cloud project id (e.g. my-project).
+     * @param zone Instance's zone.
+     * @param instanceId Instance's name.
+     *
+     * @throws IOException If an error occured in snapshot creation.
+     * @throws InterruptedException If snapshot creation is interrupted.
+     */
+    public void createSnapshot(String projectId, String zone, String instanceId) {
+        Snapshot snapshot = new Snapshot();
+        try {
+            zone = nameFromSelfLink(zone);
+
+            snapshot.setName(instanceId);
+            Operation op = compute.disks().createSnapshot(projectId, zone, instanceId, snapshot).execute();
+
+            // poll for result
+            waitForOperationCompletion(projectId, op.getName(), op.getZone(), SNAPSHOT_TIMEOUT);
+        } catch (InterruptedException ie) {
+            // catching InterruptedException here because calling function also can throw InterruptedException from trying to terminate node
```

Comment:
Do we want to be swallowing these exceptions here? What is the expected behavior when a snapshot fails?

Reply:
I think surfacing the issue for downstream callers is appropriate, as it gives them the flexibility to decide how to handle it. My suggestion would be to keep the log statements here, but then throw the exceptions.

Reply:
I'm not sure how I want to handle these exceptions when thrown, which was one of my first concerns in the first iteration of this PR.
Also, in doDoDelete, both createSnapshot and node.terminate() could both cause an InterruptedException. I'm not sure how to handle this case either.

Reply:
My suggestion would be to leave the throws in for now, and then we can take a look at how to handle them when you tackle the downstream code. You can always alter as needed later on :)

Reply:
done